### ISSUE TITLE
Disk Encryption

### ISSUE THREAD

**Issue by mjudeikis** [kubernetes-sigs/cluster-api-provider-azure#982]
***Goals***
1. Enable OSDisk and ephemeral disk encryption for Azure VMS

***Non-Goals/Future Work***
1. To enable manage Service or user keys. Encryption key management is left for the user.

**User Story**

As a User, I would like to encrypt all the disk for Azure vm/vmss for security reasons.

**Detailed Description**

Azure supports OSdisk encryption via:
`OSDisk.ManagedDisk.DiskEncryptionSet.ID` and with new compute API Host encryption too
https://docs.microsoft.com/en-us/azure/virtual-machines/disks-enable-host-based-encryption-portal 

**Contract changes [optional]**

Don't know

**Data model changes [optional]**

1. Add `ManagedDisk` option for managed disk encryption:
```
type ManagedDisk struct {
	StorageAccountType string `json:"storageAccountType"`
	StorageAccountType string                       `json:"storageAccountType"`
	DiskEncryptionSet  *DiskEncryptionSetParameters `json:"diskEncryptionSet,omitempty"`
}

// DiskEncryptionSetParameters defines disk encryption options
type DiskEncryptionSetParameters struct {
	ID string `json:"id,omitempty"`
}
```

2. Add `SecurityProfile into VM main spec (to mimic SDK configuration). Security profile adding would allow extension later on with other security-related features.
```
// AzureMachineSpec defines the desired state of AzureMachine
type AzureMachineSpec struct {
	// ProviderID is the unique identifier as specified by the cloud provider.
	// +optional
	ProviderID *string `json:"providerID,omitempty"`

	VMSize           string           `json:"vmSize"`
	......

      // SecurityProfile specifies the Security profile settings for the virtual machine or virtual machine scale
      // set.
     type SecurityProfile struct {
	  // EncryptionAtHost - This property can be used by user in the request to enable or disable the Host Encryption for the virtual machine or virtual machine scale set. This will enable the encryption for all the disks including Resource/Temp disk 
 at host itself. <br><br> Default: The Encryption at host will be disabled unless this property is set to true for the resource.
	EncryptionAtHost *bool `json:"encryptionAtHost,omitempty"`
     }

}
```

/kind proposal

**Comment by CecileRobertMichon** [comment]
@mjudeikis are you planning to work on the second part or should we mark this as help wanted?

**Comment by mjudeikis** [comment]
@dkorzuno has the second part on our side in the next sprint. It will be done :) Give us few days to tests/check-in with the right people on our end and expect for PR :) 

@dkorzuno please tag me on any future work

**Comment by CecileRobertMichon** [comment]
Should this issue be closed now that #1012 has merged @mjudeikis ?
